context/src/main/java/io/opentelemetry/context/ImplicitContextKeyed.java

```diff
@@ -21,6 +21,12 @@
    * automatically.
    *
    * <p>This method is equivalent to {@code Context.current().with(value).makeCurrent()}.
+   *
+   * <p>The default implementation of this method will store the {@link ImplicitContextKeyed} in a
+   * {@link ThreadLocal}. Kotlin coroutine users SHOULD NOT use this method as the {@link
```

Comment:
I wonder if we should also mention that many async framework users won't want to use this either. Even if we don't have a drop-in solution for them.